### PR TITLE
Use lambda to define backend operations

### PR DESCRIPTION
**Pull request by krzysztof-kwitt**

### CODE REVIEW

File: ibis/backends/mysql/registry.py [hunk 1]
```diff
@@ -147,26 +105,6 @@ def _group_concat(t, op):
     return sa.func.group_concat(arg.op('SEPARATOR')(sep))
 
 
-def _day_of_week_index(t, op):
-    left = sa.func.dayofweek(t.translate(op.arg)) - 2
-    right = 7
-    return (left % right + right) % right
```
Comment (krzysztof-kwitt) [inline ibis/backends/mysql/registry.py:153]: I simplified this function because it was overly complicated.

**Comment by cpcloud** [comment]
@krzysztof-kwitt Can you squash the commits into one commit per backend? Otherwise, this is *great*.

Let's ring in the new year with some code deletion! 🎉

**Comment by krzysztof-kwitt** [comment]
@cpcloud Squashed.

**Comment by cpcloud** [comment]
Looks like a tiny typo in the last commit, should be `refactor(duckdb):` 😂

**Review by cpcloud — APPROVED** [review]

**Comment by krzysztof-kwitt** [comment]
@cpcloud Fixed. I also noticed that two operations could be inlined in this backend.

**Comment by cpcloud** [comment]
Awesome!

**Pull request merged**